### PR TITLE
finish voting on an article

### PR DESCRIPTION
**Pull request by holyseo**

### CODE REVIEW

File: src/utils.js [hunk 1]
```diff
@@ -23,3 +23,29 @@ export const getCommentsByArticleId = (id) => {
...
+  return axios
+    .patch(`https://news-api-5rlq.onrender.com/api/articles/${id}`, {
+      inc_votes: +1,
+    })
+    .then((res) => {
+      return res.data.vote;
+    })
+    .catch((err) => {
+      return "failed to fetch counts";
+    });
```
Comment (popatre) [inline src/utils.js]: You need to handle your errors at the end of your chain, not in here. 

This message isnt great as well for a user on the website. Do they know what fetch counts means?

**Review by popatre — CHANGES_REQUESTED** [review]
- UI updates and lets us know that the votes have been incremented - nice one 👍
- Nice notifying the user when the patch fails - good UX to have  some feedback that something has gone wrong - however, please see my comment regarding your message and how you're handling the .catch

- [ ] You should ideally handle the updating of your votes using optimistic rendering.At the moment, if im on a slow internet connection its a poor user experience with the loading message

For a better UX - use optimistic rendering - 

https://notes.northcoders.com/courses/js-front-end/ui-behaviours-and-optimistic-rendering#optimistic-rendering

File: src/components/VoteArticle.jsx [hunk 2]
```diff
@@ -6,17 +6,25 @@ const VoteArticle = ({ article }) => {
...
   const handleVoteIncrement = () => {
-    setIsLoading(true);
+    typeof vote === "number"
+      ? setVote((vote) => vote + 1)
+      : setVote((vote) => +vote.replace(`Vote has been updated: `, "") + 1);
     increaseVoteById(article.article_id, vote).then((voteCount) => {
-      setVote(voteCount);
-      setIsLoading(false);
+      !voteCount
+        ? (setVote("problem with getting the counts.."),
```
Comment (nc-poppy-m) [inline src/components/VoteArticle.jsx]: Consider making this a seperate error message, not the state of the votes

File: src/utils.js [hunk 3]
```diff
@@ -33,7 +33,7 @@ export const increaseVoteById = (id, vote) => {
       return res.data.vote;
     })
     .catch((err) => {
-      return "failed to fetch counts";
+      console.log(err);
```
Comment (nc-poppy-m) [inline src/utils.js]: The catch should go in your voteArticle, not here

**Review by nc-poppy-m — CHANGES_REQUESTED** [review]
Good job on trying to implement the feedback, there's just a little ways to go:

- Instead of setting the state to the error message, why not add an element that renders an error message if there is an error? 
- Trying to add a vote when offline results in NaN votes, which looks quite buggy. It would be nice if the votes could revert to the original value
- The .catch should be where you're handling the vote increase rather than in your utils

Keep going. Try and make these changes, then resubmit :)

**Review by nc-modou — COMMENTED** [review]
Good response to feedback, site is definitely a smoother user experience now. The only thing left that you could change is that when you vote multiple times, 'Vote has been updated:' disappears then appears again which is a bit weird-looking. I would change this to make the site seem more smooth, but I suppose it's not a dealbreaker

**Pull request merged**